entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -346,6 +350,70 @@ public void bulkUpdate(
     }
   }
 
+  @Override
+  public void bulkUpdateEntityArrayAttribute(
+      BulkEntityArrayAttributeUpdateRequest request,
+      StreamObserver<ResultSetChunk> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    String tenantId = requestContext.getTenantId().orElse(null);
+    if (isNull(tenantId)) {
+      responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
+      return;
+    }
+    try {
+      Set<Key> keys =
+          request.getEntityIdsList().stream()
+              .map(entityId -> new SingleValueKey(tenantId, entityId))
+              .collect(Collectors.toSet());
```

Comment:
collect to list (or ordered set - point is to maintain order)

Reply:
ordering for keys matter only if we want to update them in a specific order. I don't think that is the case here.